deployment/terraform/vpn-ec2.tf

```diff
@@ -0,0 +1,114 @@
+data "aws_ami" "aws_ami_vpn_ec2" {
+  most_recent = true
+  owners      = ["amazon"]
+  filter {
+    name   = "architecture"
+    values = ["arm64"]
+  }
+  filter {
+    name   = "name"
+    values = ["al2023-ami-2023*"]
+  }
+}
+
+data "aws_iam_policy_document" "vpn_instance_assume_role" {
+  statement {
+    effect = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["ec2.amazonaws.com"]
+    }
+
+    actions = ["sts:AssumeRole"]
+  }
+}
+
+resource "aws_iam_role" "vpn_instance" {
+  name               = "vpn${local.short}InstanceRole"
+  assume_role_policy = data.aws_iam_policy_document.vpn_instance_assume_role.json
+}
+
+resource "aws_iam_role_policy_attachment" "vpn_instance_ssm" {
+  role       = aws_iam_role.vpn_instance.name
+  policy_arn = "arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore"
+}
+
+resource "aws_iam_instance_profile" "vpn_instance" {
+  name = aws_iam_role.vpn_instance.name
+  role = aws_iam_role.vpn_instance.name
+}
+
+# TODO: create only for RBA environment
+resource "aws_instance" "vpn_ec2" {
+  count         = var.environment == "Test" ? 1 : 0
+  ami           = data.aws_ami.aws_ami_vpn_ec2.id
+  instance_type = "t4g.nano"
+  subnet_id     = module.vpc.public_subnet_ids[count.index]
+
+  associate_public_ip_address = true
```

Comment:
We should keep `associate_public_ip_address = true` for the VPN server because:

1. The VPN server needs a public IP for WireGuard to work - clients need to be able to connect to it from the internet (to obtain key pairs for their local wg-easy setup).
2. The Elastic IP is still needed for stable WireGuard connections.
3. The security is now controlled by the security group rules, not by the presence of a public IP.